src/autoSize.js

```diff
@@ -3,31 +3,39 @@ const util = require('./util');
 const WIDTH_MIN_SIZE = 40;
 const DEBOUNCE_TIMEOUT = 200;
 
-// If element's width is less than parent width, use the parent's. If
-// resulting width is less than minimum, use the minimum. Do this to
-// Avoid failing to resize when window expands and avoid setting sizes
-// to 0 when el.offsetWidth == 0.
+/**
+ * Function that returns an element width that's equal to or greater than
+ * WIDTH_MIN_SIZE.
+ *
+ * If the element width is less than the minimum, it will recursively
+ * look for first parent element with width greater than WIDTH_MIN_SIZE.
+ *
+ * As a fallback, if the resulting width is still less than minimum, width
+ * is set to WIDTH_MIN_SIZE.
+ *
+ * We do this to avoid failing to resize when window expands and avoid
+ * setting sizes to 0 when el.offsetWidth == 0.
+ */
 const getWidth = function ({ parent, width }) {
-  // TODO: add check and test for parent == null
-  let parentWidth = parent.offsetWidth;
+  if (width < WIDTH_MIN_SIZE) {
+    width = WIDTH_MIN_SIZE;
 
-  // get the fist parent that has a size over the minimum
-  let parentNode = parent.parentNode;
-  while (parentNode && parentWidth < WIDTH_MIN_SIZE) {
-    parentWidth = parentNode.offsetWidth;
+    // TODO: add check and test for parent == null
+    let parentWidth = parent.offsetWidth;
+    let parentNode = parent.parentNode;
 
-    // set for next loop
-    parentNode = parentNode.parentNode;
-  }
+    // get the fist parent that has a size over the minimum
+    while (parentNode && parentWidth < width) {
+      parentWidth = parentNode.offsetWidth;
 
-  if (width < parentWidth) {
-    width = parentWidth;
-  }
+      // set for next loop
+      parentNode = parentNode.parentNode;
+    }
 
-  if (width < WIDTH_MIN_SIZE) {
-    width = WIDTH_MIN_SIZE;
+    if (parentWidth > width) {
+      width = parentWidth;
+    }
```

Comment:
What's changed here is that we don't just default to the largest width available. We _first_ check that the `width` argument is less than the minimum. Only then do we try and set `width` to the parent width.

Reply:
Oh this is nice change 👏